### DISCUSSION TITLE
How to install private github repository via npm through "git+https" github actions workflow ci

### DISCUSSION THREAD

**Discussion by aarefiev** [community/community#25936] [Actions]
<p>I found the same problem with installing through <code>git+https</code> (<a href="https://github.community/t/how-to-install-private-github-repository-via-npm-in-github-actions-workflow-ci/18479" class="inline-onebox">How to install private github repository via npm in github actions workflow ci</a>). According to our infrastructure requirements (we are using Docker, but build args are not supported now), I can’t create <code>id_rsa</code> on build stage :cry:.</p>
<p>According to the same topic, I got the following error on <code>npm ci</code> step:</p>
<pre><code class="lang-auto">npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ***github.com/org/private_repo.git
npm ERR! 
npm ERR! remote: Repository not found.
npm ERR! fatal: repository 'https://github.com/org/private_repo.git/' not found
npm ERR! 
npm ERR! exited with error code: 128
</code></pre>
<p><strong>ci.yml</strong></p>
<pre><code class="lang-yaml">name: Test

on:
  pull_request:
    branches:
      - master
  push:

jobs:
  test:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [10.x]
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}
      - run: npm ci
      - run: npm run lint
      - run: npm run test
</code></pre>
<p><strong>package.json</strong></p>
<pre><code class="lang-auto">"dependencies": {
  "@org/private-pkg": "git+https://token:x-oauth-basic@github.com/org/private_repo.git#tag"
}
</code></pre>
<p>This  <code>@org/private-pkg</code>  is being installed via github by npm on local machine and inside Docker. The repo is within the same organization as which the workflow is running in.</p>
<p>But how can I solve this issue with Github Actions?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/aarefiev">@aarefiev</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p><code>actions/checkout@v2</code> will persist GITHUB_TOKEN and add extra <code>http</code> header to latter git operations.</p>
<p>Please add <code>persist-credential: false</code> to <code>actions/checkout@v2</code>, and add below command before npm command:<br>
<code>run: git config --global url."https://${{ secrets.PAT }}@github.com/".insteadOf ssh://git@github.com/</code></p>
<p>Please refer to the <a href="https://stackoverflow.com/questions/62122834/install-github-repo-within-github-action-workflow" rel="nofollow noopener">link</a> for more details.</p>
<p>Thanks.</p>
